AdminCore.WebApp/Controllers/EventController.cs

```diff
@@ -161,13 +163,25 @@ public IActionResult GetEmployeeHolidayStats(int employeeId)
 
     [HttpPost]
     public IActionResult CreateEvent(CreateEventViewModel createEventViewModel)
+    {
+      return CreateEventForEmployeeId(createEventViewModel, _employee.EmployeeId);
+    }
+
+    [AdminCoreRoles(EmployeeRoles.SystemAdministrator)]
+    [HttpPost("{employeeId}")]
+    public IActionResult CreateEvent(CreateEventViewModel createEventViewModel, int employeeId)
+    {
+      return CreateEventForEmployeeId(createEventViewModel, employeeId);
+    }
+
+    private IActionResult CreateEventForEmployeeId(CreateEventViewModel createEventViewModel, int employeeId)
```

Comment:
I don't think behaviour of this function is tested...

Reply:
Both overloads CreateEvent, do have tests written for them. But I haven't written any tests for the authorization logic like what Mark was saying in the pull review.

Reply:
yep makes sense and also missed that it's a private function